code/game/gamemodes/trifecta/trifecta.dm

```diff
@@ -0,0 +1,115 @@
+	restricted_jobs = list("Cyborg")
+	secondary_restricted_jobs = list("AI")
+	required_players = 10
+	required_enemies = 1	// how many of each type are required
+	recommended_enemies = 3
+	secondary_protected_species = list("Machine")
+	var/list/datum/mind/pre_traitors = list()
+	var/list/datum/mind/pre_changelings = list()
+	var/list/datum/mind/pre_vampires = list()
+	var/amount_vamp = 1
+	var/amount_cling = 1
+	var/amount_tot = 1
+
+/datum/game_mode/trifecta/announce()
+	to_chat(world, "<B>The current game mode is - Trifecta</B>")
+	to_chat(world, "<B>Vampires, traitors, and changelings, oh my! Stay safe as these forces work to bring down the station.</B>")
+
+
+/datum/game_mode/trifecta/pre_setup()
+	calculate_quantities()
+	if(GLOB.configuration.gamemode.prevent_mindshield_antags)
+		restricted_jobs += protected_jobs
+	var/list/datum/mind/possible_vampires = get_players_for_role(ROLE_VAMPIRE)
+
+	if(!length(possible_vampires))
+		return FALSE
+
+	if(length(possible_vampires))
+		for(var/I in possible_vampires)
+			if(length(pre_vampires) >= amount_vamp)
+				break
+			var/datum/mind/vampire = pick_n_take(possible_vampires)
+			if(vampire.current.client.prefs.active_character.species in secondary_protected_species)
+				continue
+			pre_vampires += vampire
+			vampire.special_role = SPECIAL_ROLE_VAMPIRE
+			vampire.restricted_roles = (restricted_jobs + secondary_restricted_jobs)
+
+	//Vampires made, off to changelings
+	var/list/datum/mind/possible_changelings = get_players_for_role(ROLE_CHANGELING)
+
+	if(!length(possible_changelings))
+		return FALSE
+
+	for(var/I in possible_changelings)
+		if(length(pre_changelings) >= amount_cling)
+			break
+		var/datum/mind/changeling = pick_n_take(possible_changelings)
+		if((changeling.current.client.prefs.active_character.species in secondary_protected_species) || (changeling.special_role == SPECIAL_ROLE_VAMPIRE))
+			continue
+		pre_changelings += changeling
+		changeling.restricted_roles = (restricted_jobs + secondary_restricted_jobs)
+		changeling.special_role = SPECIAL_ROLE_CHANGELING
+
+	//And now traitors
+	var/list/possible_traitors = get_players_for_role(ROLE_TRAITOR)
+
+	// stop setup if no possible traitors
+	if(!length(possible_traitors))
+		return FALSE
+
+	for(var/i in 1 to amount_tot)
+		if(!length(possible_traitors))
+			break
+		var/datum/mind/traitor = pick_n_take(possible_traitors)
+		if(traitor.special_role == SPECIAL_ROLE_VAMPIRE || traitor.special_role == SPECIAL_ROLE_CHANGELING) // no traitor vampires or changelings
+			continue
+		pre_traitors += traitor
+		traitor.special_role = SPECIAL_ROLE_TRAITOR
+		traitor.restricted_roles = restricted_jobs
+
+	if(!length(pre_traitors))
+		return FALSE
+	return TRUE
+
+/datum/game_mode/trifecta/proc/calculate_quantities()
+	var/points = num_players()
+	while(points > 0)
+		if(points < TOT_COST)
+			amount_tot++
+			points = 0
```

Comment:
Is it fine to not `return` after settings points to 0? Currently, the code will continue on and still pick another tot, cling, or vamp.

Reply:
will return